snuba/clickhouse/native.py

```diff
@@ -219,8 +219,12 @@ def __transform_result(self, result, with_totals: bool) -> Result:
 
     def execute(
         self,
-        query: ClickhouseQuery,
-        # TODO: move Clickhouse specific arguments into DictClickhouseQuery
+        # TODO: After we remove DictSqlQuery (that does all query processing for the
+        # legacy query representation) we should pass the Clickhouse Query here. and the reader
+        # should rely on the formatter to get a SQL string and, thus, be able to provide parameters
+        # to the formatter (like the format clause itself).
```

Comment:
Thinking about it — this probably means that the `FORMAT` clause is _not_ part of `ClickhouseQuery` (since its only pertinent to the `Reader`?)

Reply:
it seems to me that the reader implementation is the only one who cares about the output format. At least I thought this was the reason and that it was tied to the HTTP Reader.
Was there any other reason to make format part of the Clickhosue Query ?

Reply:
> Was there any other reason to make format part of the Clickhosue Query ?

Not that I can think of. (This was more of a clarifying question than one that was supposed to imply a change in direction.)